### PR TITLE
🚀Preview Release

### LINKED ISSUE #501: 🚀Preview Release
### Pre-Release ToDo List

- [X] All issues in the assigned milestone are closed and all issue tasks are complete.
- [X] Add `🚀Preview Release` label to this issue.
- [X] Issue is assigned to a project.
- [X] Issue is assigned to a milestone.
- [x] Release PR linked to this issue.
- [x] All unit tests have been executed locally and have passed. _(Check out the appropriate release branch before running tests)_.
- [x] Version in project file updated _(All changes made directly on the release branch (preview/v#.#.#-preview.#))_.
- [x] Release notes created and added _(All changes made directly on the release branch (preview/v#.#.#-preview.#))_.
- [x] Manual QA Testing completed _(if applicable)_.
- [x] Pull request has been approved and merged into a release branch before performing the release. _(Releases are performed on preview branches)_.
- [ ] Release to **_preview_** complete. _(The release is performed by running the `🚀Preview Release` workflow)_.

### Post-Release ToDo List

- [ ] The GitHub release has been created and is correct.
- [ ] The NuGet package has been successfully deployed to [nuget.org](https://www.nuget.org/) _(if applicable)_.
- [X] Announcement of release on [Twitter](https://twitter.com/KDCoder) verified _(if applicable)_. _(Announcement should be performed automatically with the release)_.
- [X] Announcement has been pushed to the [Discord](https://discord.gg/qewu6fNgv7) channel _(if applicable)_.

### Additional Information:

**_<details closed><summary>Unit Tests</summary>_**

Reasons for local unit test execution:
- Unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
- Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
</details>

---

**_<details closed><summary>Version Updating</summary>_**

The version can be updated by setting the values of the `<Version/>` and `<FileVersion/>` XML tags in the project file.
The `<Version/>` and `<FileVersion/>` values can hold production and preview releases.
The `<AssemblyVersion/>` XML tag can only hold production values.  Preview values are not allowed.

``` xml
<!--Preview Release Example-->
<Version>1.2.3-preview.4</Version>
<FileVersion>1.2.3-preview.4</FileVersion>
<AssemblyVersion>1.2.3</AssemblyVersion>
```
</details>

---

**_<details closed><summary>Release Notes</summary>_**

By default, the release notes go into a **_ReleaseNotes_** folder inside of the **_Documentation_** folder with the **_Documentation_** folder at the root of the repository.
Preview release notes path: **_./Documentation/ReleaseNotes/PreviewReleases_**.

Release note file names must follow a particular syntax and are in markdown format so they can be added to the release.

#### Release Notes File Name Syntax:
- **Preview Release Notes:**
  - **Syntax:** Release-Notes-v#.#.#-preview.#.md
  - **Example:** Release-Notes-v1.2.3-preview.4.md

Changes such as release notes and version updates should be committed to the destination branch (preview/v#.#.#-preview.#) in the pull request attached to this issue.
</details>

---

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Release Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

- [x] The **_base(target)_** branch for this pull request is a **_release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: release/v<major>.<minor>.<patch>
      Example: release/v1.2.3
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_release_** branch with the correct naming syntax in the following example.
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: release/v<major>.<minor>.<patch>
    Example: release/v1.2.3
    ```
  </details>

💡For more information on branching, refer to the project's [branching documentation](../../Documentation/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title is set to _**🚀Preview Release**_.

- [x] A `🚀Preview Release` label has been added to this pull request. 

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] A QA issue has been created and manual QA testing has been performed for the changes being released _(if applicable)_.

### CODE REVIEW

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+   - UI controls are now always rendered on top of textures, text, lines, and rectangles.
+     >💡There are plans in the future to add control to this behavior.
+
+---
+
+<h2 style="font-weight:bold" align="center">Breaking Changes 🧨</h2>
+
+1. [#485](https://github.com/KinsonDigital/Velaptor/issues/451) - Added the breaking changes listed below:
+   - Changed the numerical value of the `RenderEffects` enum.
+     >💡This was not a change the enum names.  This was required to facilitate empty checks for internal render items.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md:29]: Grammar
```suggestion
     >💡This was not a change to the enum names.  This was required to facilitate empty checks for internal render items.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+   - Change the exception thrown by the `PathResolverFactory.CreateSystemFontPathResolver()` method when executed on non-windows platforms.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md:30]: Grammar
```suggestion
   - Changed the exception thrown by the `PathResolverFactory.CreateSystemFontPathResolver()` method when executed on non-windows platforms.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+   - Removed the method `CreateRenderer()` from the `RendererFactory` class.
+     >💡This has been replaced with multiple methods to create different render types.
+   - Changed the `RendererFactory` to `sealed`.
+   - There is not a single renderer that is used to render everything anymore.  This has been removed and replaced by the different rendering types below:
+     >💡These are created using the `RendererFactory` class.
+     - `ITextureRenderer`
+     - `IFontRenderer`
+     - `IRectangleRenderer`
+     - `ILineRenderer`
+   - The process for beginning and ending a batch has now changed. This is now done by using the `Begin()` and `End()` methods in the `IRenderer` interface.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md:42]: Reword
```suggestion
   - The process for beginning and ending a batch changed. This is now done by using the `Begin()` and `End()` methods in the `IRenderer` interface.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+   - Changed the name of the `CreateFontAtlas()` method to `CreateAtlas()` in the `IFontAtlasService` interface.
+   - Changed the name of the `textureAtlas` parameter to `atlasTexture` in the `Create()` method in the `IFontFactory` interface.
+   - Changed the class `AtlasSubTextureData` to a `readonly struct`.
+
+---
+
+<h2 style="font-weight:bold" align="center">Internal Changes ⚙️</h2>
+<h5 align="center">(Changes that do not affect users.  Not breaking changes, new features, or bug fixes.)</h5>
+
+1. [#485](https://github.com/KinsonDigital/Velaptor/issues/451) - Greatly improve internal batching of render items which as a result greatly improved performance.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md:59]: Reword and make sure correct issue is linked. 
```suggestion
1. [#485](https://github.com/KinsonDigital/Velaptor/issues/485) - Improved internal batching of render items which has greatly improved performance.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+2. [#497](https://github.com/KinsonDigital/Velaptor/issues/497) - Improved performance by changing how buffering on the GPU is handled.
+
+---
+
+<h2 style="font-weight:bold" align="center">Nuget/Library Updates 📦</h2>
+
+1. [#485](https://github.com/KinsonDigital/Velaptor/issues/451) - Updated **KinsonDigital.Carbonate** from _**v1.0.0-preview.12**_ to _**v1.0.0-preview.14**_
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md:66]: Incorrect link - should be 485.  
```suggestion
1. [#485](https://github.com/KinsonDigital/Velaptor/issues/485) - Updated **KinsonDigital.Carbonate** from _**v1.0.0-preview.12**_ to _**v1.0.0-preview.14**_
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+<div align="center">
+
+As with all software, there is always a chance for issues and bugs, especially for preview releases, which is why your input is greatly appreciated. 🙏🏼
+</div>
+
+---
+
+<h2 style="font-weight:bold" align="center">New Features ✨</h2>
+
+1. [#485](https://github.com/KinsonDigital/Velaptor/issues/451) - Improved layered rendering in the following ways:
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md:16]: Refactor to fix link.  It's incorrect throughout the whole document.
```suggestion
1. [#485](https://github.com/KinsonDigital/Velaptor/issues/485) - Improved layered rendering in the following ways:
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+3. [#479](https://github.com/KinsonDigital/Velaptor/issues/479) - Updated all uses of the GitHub [checkout](https://github.com/marketplace/actions/checkout) action from _**v2**_ to _**v3**_ in all of the workflows.
+4. [#492](https://github.com/KinsonDigital/Velaptor/issues/492) - Added API documentation to the nuget package.
+   >💡This will provide API documentation in intellisense in your preferred IDE.
+5. [#477](https://github.com/KinsonDigital/Velaptor/issues/477) - Improved grammar and spelling issues in previous preview release notes.
+6. [#377](https://github.com/KinsonDigital/Velaptor/issues/377) - Moved the setup to expose the visibility of `internal`` types to the unit test project to the **Velaptor** project file.
+7. [#361](https://github.com/KinsonDigital/Velaptor/issues/361) - Improved unit tests by refactoring assertion code to use [Fluent Assertions](https://fluentassertions.com/).
+8. [#152](https://github.com/KinsonDigital/Velaptor/issues/152) - Setup and the project to automatically publish unit test code coverage results to [codecov](https://about.codecov.io/).
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.16.md:85]: Grammar
```suggestion
8. [#152](https://github.com/KinsonDigital/Velaptor/issues/152) - Setup the project to automatically publish unit test code coverage results to [codecov](https://about.codecov.io/).
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by kselena — DISMISSED** [review]

**Review by KinsonDigitalAdmin — DISMISSED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Pull request merged**